lib/auth/windows/windows.go

```diff
@@ -107,6 +107,13 @@ func getCertRequest(req *GenerateCredentialsRequest) (*certRequest, error) {
 		csr.ExtraExtensions = append(csr.ExtraExtensions, createUser)
 	}
 
+	if req.AD {
+		csr.ExtraExtensions = append(csr.ExtraExtensions, pkix.Extension{
```

Comment:
This means if the extension is missing, then we assume it's non-AD, right?

Should we invert that? We supported AD first, so are there advantages to requiring an explicit flag for non-AD and treating a missing extension as AD?